sources/packages/web/src/components/aest/students/modals/EditStudentProfileModal.vue

```diff
@@ -0,0 +1,182 @@
+          ><template #value
+            ><v-text-field
+              hide-details="auto"
+              label="Email"
+              v-model="formModel.email"
+              variant="outlined"
+              :rules="[checkEmailLengthRule, checkEmailValidationRule]"
+              class="mb-6" />
+            <v-spacer /></template
+        ></title-value>
+        <title-value
+          ><template #value>
+            <v-textarea
+              label="Notes"
+              v-model="formModel.noteDescription"
+              variant="outlined"
+              :rules="[checkNotesLengthRule]" />
+            <v-spacer /></template
+        ></title-value>
+        <title-value
+          ><template #value
+            ><v-input
+              :rules="[hasDataChangedValidationRule]"
+              hide-details="auto"
+            /> </template
+        ></title-value>
+      </template>
+      <template #footer>
+        <footer-buttons
+          :processing="loading"
+          primaryLabel="Update Profile"
+          @primaryClick="submit"
+          @secondaryClick="cancel"
+        />
+      </template>
+    </modal-dialog-base>
+  </v-form>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref } from "vue";
+import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
+import TitleValue from "@/components/generic/TitleValue.vue";
+import { useFormatters, useModalDialog, useRules } from "@/composables";
+import {
+  UpdateStudentDetailsAPIInDTO,
+  UpdateStudentDetails,
+} from "@/services/http/dto";
+import { Role, VForm } from "@/types";
+
+export default defineComponent({
+  components: {
+    ModalDialogBase,
+    TitleValue,
+  },
+  setup() {
+    const {
+      checkGivenNameLengthRule,
+      checkLastNameLengthRule,
+      checkEmailLengthRule,
+      checkEmailValidationRule,
+      checkNotesLengthRule,
+    } = useRules();
+    const { getISODateOnlyString } = useFormatters();
+    const getTodaysDate = (): string => {
+      return getISODateOnlyString(new Date());
+    };
+    const {
+      showDialog,
+      showModal: showModalInternal,
+      resolvePromise,
+      loading,
+    } = useModalDialog<UpdateStudentDetailsAPIInDTO | boolean>();
+    const editStudentProfileForm = ref({} as VForm);
+    const formModel = ref({} as UpdateStudentDetailsAPIInDTO);
+    const submit = async () => {
+      const validationResult = await editStudentProfileForm.value.validate();
+      if (!validationResult.valid) {
+        return;
+      }
+      await resolvePromise(formModel.value);
+    };
+    let initialStudentProfile = {} as UpdateStudentDetails;
+
+    const hasDataChangedValidationRule = () => {
+      if (
+        initialStudentProfile.email === formModel.value.email &&
+        initialStudentProfile.givenNames === formModel.value.givenNames &&
+        initialStudentProfile.lastName === formModel.value.lastName &&
+        initialStudentProfile.birthdate === formModel.value.birthdate
```

Comment:
This validation should be aligned with the API. Please add the case insensitive check also.